cpp/src/binaryop/binaryop.cpp

```diff
@@ -253,7 +253,7 @@ std::unique_ptr<column> make_fixed_width_column_for_output(scalar const& lhs,
   if (binops::is_null_dependent(op)) {
     return make_fixed_width_column(output_type, rhs.size(), mask_state::ALL_VALID, stream, mr);
   } else {
-    auto new_mask = binops::scalar_col_valid_mask_and(rhs, lhs, stream, mr);
+    auto new_mask = binops::scalar_col_bitmask_and(rhs, lhs, stream, mr);
     return make_fixed_width_column(
       output_type, rhs.size(), std::move(new_mask), cudf::UNKNOWN_NULL_COUNT, stream, mr);
```

Comment:
This will also get a null count once the API is refactored above. I think it should look like:
```suggestion
    auto [new_mask, new_null_count] = binops::scalar_col_bitmask_and(rhs, lhs, stream, mr);
    return make_fixed_width_column(
      output_type, rhs.size(), std::move(new_mask), new_null_count, stream, mr);
```